### PR TITLE
doc: add interacting with flux section

### PR DESCRIPTION
**Pull request by garlick**
Problem: flux-core documentation explains various ways to start a Flux instance but doesn't provide insight into how to connect to them or what to do then.

Add an "interacting with flux" section.

Also add a mostly empty glossary section so we'll have it ready to be expanded as more detail is added.

### CODE REVIEW

File: doc/guide/interact.rst [hunk 1]
```diff
@@ -0,0 +1,354 @@
...
+
+       View the progress of running jobs.  See :man1:`flux-top`.
+
+   * - ``flux batch -N4 script.sh``
+
+       ``flux batch -N1 --wrap sleep 60``
+
+       ``flux pstree``
+
+     - Submit job to run script.sh in 4 node instance.  See :man1:`flux-batch`.
```
Comment (chu11) [inline doc/guide/interact.rst:112]: "in a 4 node instance"?

File: doc/guide/interact.rst [hunk 2]
```diff
@@ -0,0 +1,354 @@
...
+
+       ``flux pgrep -f pending .``
+
+       ``flux pkill sl..p``
+
+     - Cancel specified job.  See :man1:`flux-cancel`.
+
+       Send specified job a SIGHUP.  See :man1:`flux-job`.
+
+       List ids of all pending jobs.
```
Comment (chu11) [inline doc/guide/interact.rst:146]: Forgot to add "see flux-grep(1)"?

File: doc/guide/interact.rst [hunk 3]
```diff
@@ -0,0 +1,354 @@
...
+
+       Cancel all jobs named sleep or slurp.  See :man1:`flux-pgrep`.
```
Comment (chu11) [inline doc/guide/interact.rst:148]: i think this one should be "See flux-pkill(1)"

File: doc/guide/interact.rst [hunk 4]
```diff
@@ -0,0 +1,354 @@
...
+  $ FLUX_URI=$(flux uri $(flux job last)) flux jobs
+       JOBID USER     NAME       ST NTASKS NNODES     TIME INFO
+    ƒ3ctHRVy alice    sleep       S      2      2        -
+    ƒ3ctHRVz alice    sleep       S      2      2        -
+    ƒ3cumQnK alice    sleep       S      2      2        -
+    ƒ3cwFQ4f alice    sleep       S      2      2        -
+    ƒ3croSDd alice    sleep       R      2      2   6.593s test[3-4]
+
+That started a batch job with a lifetime of 2m, then submitted 5 "sleep 60"
+jobs to it, then listed the batch job's active jobs.
```
Comment (chu11) [inline doc/guide/interact.rst:223]: it took me awhile to figure out how it had a lifetime of 2m, i was looking for a `--time-limit`, perhaps just say "120 seconds".

**Review by chu11 — APPROVED** [review]
overall LGTM, just a few things I noticed

**Comment by garlick** [comment]
Thanks!  Fixed those things and also added another item to the glossary.

**Comment by garlick** [comment]
Rebased on current master and forced a push.

**Pull request merged**